### PR TITLE
Fix basic auth on OAuth2  client credentials flow

### PR DESCRIPTION
**Pull request by peterhartman**
It looks like `init_oauth2.0` is not passing `use_basic_auth` onwards. I tried (and failed) to create a test for the client credentials flow using httpbin. I can't find a public API suitable for testing against unfortunately.

### CODE REVIEW

**Comment by peterhartman** [comment]
I finally managed to create a test using a dummy vimeo user. Phew!

I have confirmed that the test fails without my proposed change.

**Comment by chicofish** [comment]
Hi Peter,

I'm having the same issue and I've got to get something moving here at work.  Is there something that I can do in my R code to get this to work while waiting for the package to be updated?

Thanks

Fred

**Comment by cderv** [comment]
You should use `devtools::install_github("r-lib/httr#485")` to install this PR which have the fix you want.
It will install `httr` in this state and will work until it is merged and on CRAN.

**Comment by chicofish** [comment]
Thank you.  I did that and it has not corrected my issue, which is a bad request (400 error) and not getting a token.  I had thought that this was a solution to that issue.  sigh.

Error in oauth2.0_access_token(endpoint, app, code = code, user_params = user_params,  : 
  Bad Request (HTTP 400). Failed to get an access token.

Facebook acts as though the request was successful in the browser, but no dice in my R session.

Any thoughts would be helpful but I understand it's probably beyond the scope of this thread.

**Comment by cderv** [comment]
Did you follow [httr facebook demo](https://github.com/r-lib/httr/blob/master/demo/oauth2-facebook.r) ? 
There is a [SO question](https://stackoverflow.com/questions/37577908/how-to-use-the-oauth2-0-token-function-from-httr-package-with-facebook-api) which could be related to. 

If still an issue, you should open a post on SO with a reproductible example rather than here as it is not a bug but a configuration problem.

**Comment by chicofish** [comment]
I did finally find my problem, thank you.  It was a configuration on the Facebook app side.

httr is working wonderfully.  Thanks for your assistance.

**Comment by GregoireCoppens** [comment]
Thank you!
We had this problem in #482 (reddit API giving 401 error) and your version fixes this perfectly!

**Comment by peterhartman** [comment]
@hadley Apologies for the directness, how do I request a review of this PR?

**Comment by peterhartman** [comment]
@jimhester I see you were able to request a review for #507, does this PR seem worth considering? I am happy to put more work in if something is missing.

**Comment by hadley** [comment]
@peterhartman httr is on my to do list for after TG

File: tests/testthat/test-oauth.R [hunk 1]
```diff
@@ -25,6 +25,34 @@ test_that("oauth2.0 signing works", {
   )
 })
 
+test_that("oauth2.0 basic auth on client credentials works", {
```
[the remark this answers is not in the document]
Comment (hadley) [reply tests/testthat/test-oauth.R]: i.e. fix #491

**Comment by hadley** [comment]
Thanks for your help with this @peterhartman!

**Pull request merged**